src/main/java/org/openjdk/jextract/impl/HeaderFileBuilder.java

```diff
@@ -170,6 +168,44 @@ class Holder {
                 }
             }
             """);
+        } else {
+            String invokerName = javaName + "$invoker";
+            String invokerFactoryName = javaName + "$makeInvoker";
+            String paramExprs = paramExprs(declType, finalParamNames, isVarArg);
+            appendLines(STR."""
+                public interface \{invokerName} {
```

Comment:
question: if we used a record here, could we perhaps get rid of the holder class (since record fields are trusted) ? We don't expect this interface to be created by the user, so there's maybe not much value in it being a functional interface?

Reply:
Not sure what you mean. The holder class is for the base descriptor. We could stick that into the invoker interface instead, but I'm not sure why a record would be better than an interface? Do we want users to be able to access the underlying MethodHandle?

Reply:
Sorry, I guess what I'm trying to say is that:
* the benefit for caching the base descriptor is low;
* the real benefit would be in treating the final (specialized) method handle as a constant

To do the latter, given we can't use stable fields, the only way to get there is to use a record. So, if we care about maximum performance, we should use a record. If we don't care, an interface is fine. In both cases, I don't think there's much to gain in having an extra holder class here.

Reply:
Okay, I can drop the holder class.